test/extensions/transport_sockets/alts/noop_transport_socket_callbacks_test.cc

```diff
@@ -0,0 +1,57 @@
+#include "extensions/transport_sockets/alts/noop_transport_socket_callbacks.h"
+
+#include "test/mocks/network/mocks.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace TransportSockets {
+namespace Alts {
+namespace {
+
+class TestTransportSocketCallbacks : public Network::TransportSocketCallbacks {
+public:
+  TestTransportSocketCallbacks(Network::Connection* connection) : connection_(connection) {}
```

Comment:
if connection_ couldn't be null, make it a reference. also make constructor explicit